File: accounts/accounts-domain/src/main/java/com/hrrm/famoney/accounts/movement/Entry.java
package com.hrrm.famoney.accounts.movement;

import lombok.*;
import lombok.experimental.Accessors;
import org.eclipse.persistence.annotations.JoinFetch;
import org.eclipse.persistence.annotations.JoinFetchType;
import org.jetbrains.annotations.NotNull;

import javax.persistence.*;
import java.util.List;

@Entity
@DiscriminatorValue("entry")
@Accessors(chain = true)
@Getter
@Setter
@ToString(callSuper = true)
@EqualsAndHashCode(
        onlyExplicitlyIncluded = true,
        callSuper = true
)
public class Entry extends Movement {

    @ElementCollection
    @CollectionTable(
            name = "entry_item",
            joinColumns = @JoinColumn(name = "entry_id")
    )
    @JoinFetch(JoinFetchType.INNER)
    @NotNull
    private List<EntryItem> entryItems;

}

File: accounts/accounts-domain/src/main/kotlin/com/hrrm/famoney/accounts/movement/Entry.kt
package com.hrrm.famoney.accounts.movement

import org.eclipse.persistence.annotations.JoinFetch
import org.eclipse.persistence.annotations.JoinFetchType
import javax.persistence.*

@Entity
@DiscriminatorValue("entry")
class Entry : Movement() {
    @ElementCollection
    @CollectionTable(name = "entry_item", joinColumns = [JoinColumn(name = "entry_id")])
    @JoinFetch(JoinFetchType.INNER)
    val entryItems: List<EntryItem> = mutableListOf()
}